release-content/0.14/release-notes/11426_Computed_State__Sub_States.md

```diff
@@ -1 +1,107 @@
-TODO
+Bevy's `States` are a simple but powerful abstraction for managing the control flow of your app.
+
+```rust
+use bevy_state::prelude::*;
+use bevy_ecs::prelude::IntoSystemConfigs;
+use bevy_ecs::system::ResMut;
+
+
+#[derive(Clone, Copy, PartialEq, Eq, Hash, Debug, Default, States)]
+enum GameState {
+ #[default]
+  Menu,
+  InGame,
+}
+
+fn handle_escape_pressed(mut next_state: ResMut<NextState<GameState>>) {
+    if escape_pressed {
+        next_state.set(GameState::Menu);
+    }
+}
+
+fn open_menu() {
+    // Show the settings menu...
+}
+
+App::new()
+    .init_state::<GameState>()
+    .add_systems(Update, handle_escape_pressed.run_if(in_state(GameState::Menu)));
+    .add_systems(OnEnter(GameState::Menu), open_settings_menu);
+```
+
+But as users' games (and non-game applications!) grew in complexity, their limitations became more apparent.
+What happens if we want to capture the notion of "in a menu", but then have different states corresponding to which submenu should be open?
+What if we want to ask questions like "is the game paused", but that question only makes sense while we're within a game?
+
+Finding a good abstraction for this required [several](https://github.com/bevyengine/bevy/pull/9957) [attempts](https://github.com/bevyengine/bevy/pull/10088) and a great deal of both experimentation and discussion.
+
+While your existing states code will work exactly as before, there are now two additional tools you can reach for if you're looking for more expressiveness: **computed states** and **sub states**.
+
+```rust
+/// Let's begin with our simple state above
+#[derive(States, Clone, PartialEq, Eq, Hash, Debug, Default)]
+enum GameState {
+    #[default]
+    Menu,
+    // The addition of this field means that simply checking for `GameState::InGame` doesn't work:
+    // the values are different based on its value and we may want to distinguish between game systems that run
+    // when the game is paused or not!
+    InGame {
```

Comment:
I think it's a little confusing to have this enum case share a name with our derived state example. Maybe we could rename this to `Running` or something?